pkg/manager/leaderelection.go

```diff
@@ -52,7 +73,41 @@ func (k *KubeMacPoolManager) AddLeaderLabelToElectedPod() error {
 		return errors.Wrap(err, "failed to update leader label to elected kubemacpool manager pod")
 	}
 
-	log.Info("marked this manager as leader for webhook", "podName", k.podName)
+	logger.V(1).Info("marked this manager as leader for webhook", "podName", k.podName)
+	return nil
+}
+
+func (k *KubeMacPoolManager) removeLeaderLabelFromAllKubemacpoolPods() error {
```

Comment:
If we continue with this approach, we can filter by label instead of namespace so we return the pod with leader label and remove it.

Reply:
yes, but technically could be super wierd snecario where we have more than 1... I think doing it in 1 loop is better like you said in above comment